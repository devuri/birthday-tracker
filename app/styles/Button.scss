button {
  background-color: #F7E74A;
  border-radius: 5px;
  margin: 10px;
  padding: 5px;
  width: 80px;
  shadow-color: #000000;
  shadow-opacity: 0.3;
  shadow-radius: 3;
  shadow-offset {
   height: 0;
   width: 0;
 }
}

disabledButton {
  background-color: #eff1a1;
  border-radius: 5px;
  margin: 10px;
  padding: 5px;
  width: 80px;
  shadow-color: #000000;
  shadow-opacity: 0.3;
  shadow-radius: 3;
  shadow-offset {
   height: 0;
   width: 0;
 }
}

buttonText {
  color: #02556D;
  font-size: 18px;
  text-align: center;
}

buttonRow {
  flex-direction: row;
  justify-content: center;
}
