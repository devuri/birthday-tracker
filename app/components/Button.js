'use strict';
import React, { Component } from 'react';
import styles from '../styles/main';
import { Text, TouchableHighlight } from 'react-native';

const Button = ({ onPress, title }) => {
  return (
    <TouchableHighlight
      style={styles.button}
      underlayColor={'#eff1a1'}
      onPress={onPress}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableHighlight>
  );
}

export default Button;
