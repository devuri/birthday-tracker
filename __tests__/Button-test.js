import 'react-native';
import React from 'react';
import Button from '../app/components/Button.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  let title='Save';
  const tree = renderer.create(
    <Button title={title} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
