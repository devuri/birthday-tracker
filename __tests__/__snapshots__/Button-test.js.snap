exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "backgroundColor": "#F7E74A",
        "borderRadius": 5,
        "margin": 10,
        "padding": 5,
        "shadowColor": "#000000",
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
        "width": 80,
      },
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#02556D",
        "fontSize": 18,
        "textAlign": "center",
      }
    }>
    Save
  </Text>
</View>
`;
